### PR TITLE
release, ci: split builds by platform

### PR DESCRIPTION
**Pull request by rail**
Previously, Make and Publish Build, Publish Bleeding Edge, and the release build built all platforms in a single job. As a result the end-to-end time was more than one hour. This is a bit slow for an effective feedback loop.

This PR adds support to build and publish binaries in parallel by adding `--platform` and argument to `publish-artifacts` and `publish-provisional-artifacts`. Also, it modifies the wrapper scripts to address the change.

As a result the end-to-end time dropped from 1-1:20h to ~20m for the mentioned jobs.

* Rename the scripts by adding `-per-platform` suffix for easier handling the change in TeamCity.
* Added scripts to generate multi-arch docker images as a separate job.

Epic: none
Release note: None

### CODE REVIEW

**Comment by rail** [comment]
Dry-run release build: https://teamcity.cockroachdb.com/buildConfiguration/Internal_Cockroach_Release_Process_BuildAndSignCockroachRelease/12902883?buildTab=dependencies&mode=timeline&type=snapshot&range=1701287035000-1701288306000
Customized build: https://teamcity.cockroachdb.com/buildConfiguration/Internal_Cockroach_Release_Customized_MakeAndPublishCustomizedBuild/12902802?buildTab=dependencies&type=snapshot&mode=timeline&range=1701286799000-1701290359000

**Review by jlinder — APPROVED** [review]
One suggestion.

File: build/teamcity/cockroach/post-merge/publish-bleeding-edge-per-platform.sh [hunk 1]
```diff
@@ -12,13 +12,14 @@ source "$dir/teamcity-bazel-support.sh"
...
-BAZEL_SUPPORT_EXTRA_DOCKER_ARGS="-e TC_BUILDTYPE_ID -e TC_BUILD_BRANCH -e gcs_credentials -e gcs_bucket=$gcs_bucket" run_bazel << 'EOF'
+BAZEL_SUPPORT_EXTRA_DOCKER_ARGS="-e TC_BUILDTYPE_ID -e TC_BUILD_BRANCH -e gcs_credentials -e gcs_bucket=$gcs_bucket -e platform=$platform" run_bazel << 'EOF'
 bazel build --config ci //pkg/cmd/publish-artifacts
 BAZEL_BIN=$(bazel info bazel-bin --config ci)
 export google_credentials="$gcs_credentials"
 source "build/teamcity-support.sh"  # For log_into_gcloud
 log_into_gcloud
 export GOOGLE_APPLICATION_CREDENTIALS="$PWD/.google-credentials.json"
-$BAZEL_BIN/pkg/cmd/publish-artifacts/publish-artifacts_/publish-artifacts --gcs-bucket="$gcs_bucket"
+$BAZEL_BIN/pkg/cmd/publish-artifacts/publish-artifacts_/publish-artifacts --gcs-bucket="$gcs_bucket" --platform=$platform
```
[the remark this answers is not in the document]
Comment (rail) [reply build/teamcity/cockroach/post-merge/publish-bleeding-edge-per-platform.sh:24]: Quoting won't hurt. 
FTR, the value is never set to `PLATFORM must be specified` - that a way to error with a message.

```
$ PLATFORM=123 bash -xe <<< 'platform="${PLATFORM:?PLATFORM must be specified}"'
+ platform=123
$ bash -xe <<< 'platform="${PLATFORM:?PLATFORM must be specified}"'
bash: line 1: PLATFORM: PLATFORM must be specified
```

**Comment by rail** [comment]
bors r=jlinder

**Pull request merged**